Common/src/config_structure.cpp

```diff
@@ -1490,43 +1497,16 @@ void CConfig::SetConfig_Options() {
-  /*!\brief RESIDUAL_REDUCTION \n DESCRIPTION: Residual reduction (order of magnitude with respect to the initial value)\n DEFAULT: 3.0 \ingroup Config*/
-  addDoubleOption("RESIDUAL_REDUCTION_BGS_FLOW", OrderMagResidual_BGS_F, 3.0);
-  /*!\brief RESIDUAL_MINVAL\n DESCRIPTION: Min value of the residual (log10 of the residual)\n DEFAULT: -8.0 \ingroup Config*/
-  addDoubleOption("RESIDUAL_MINVAL_BGS_FLOW", MinLogResidual_BGS_F, -8.0);
-  /*!\brief RESIDUAL_REDUCTION \n DESCRIPTION: Residual reduction (order of magnitude with respect to the initial value)\n DEFAULT: 3.0 \ingroup Config*/
-  addDoubleOption("RESIDUAL_REDUCTION_BGS_STRUCTURE", OrderMagResidual_BGS_S, 3.0);
-  /*!\brief RESIDUAL_MINVAL\n DESCRIPTION: Min value of the residual (log10 of the residual)\n DEFAULT: -8.0 \ingroup Config*/
-  addDoubleOption("RESIDUAL_MINVAL_BGS_STRUCTURE", MinLogResidual_BGS_S, -8.0);
-  /* DESCRIPTION: FEM: UTOL = norm(Delta_U(k)) / norm(U(k)) */
-  addDoubleOption("RESIDUAL_FEM_UTOL", Res_FEM_UTOL, -9.0);
-  /* DESCRIPTION: FEM: RTOL = norm(Residual(k)) / norm(Residual(0)) */
-  addDoubleOption("RESIDUAL_FEM_RTOL", Res_FEM_RTOL, -9.0);
-  /* DESCRIPTION: FEM: ETOL = Delta_U(k) * Residual(k) / Delta_U(0) * Residual(0) */
-  addDoubleOption("RESIDUAL_FEM_ETOL", Res_FEM_ETOL, -9.0);
-  /* DESCRIPTION: FEM: ETOL = Delta_U(k) * Residual(k) / Delta_U(0) * Residual(0) */
-  addEnumOption("RESIDUAL_CRITERIA_FEM", Res_FEM_CRIT, ResFem_Map, RESFEM_RELATIVE);
-  /*!\brief RESIDUAL_FUNC_FLOW\n DESCRIPTION: Flow functional for the Residual criteria\n OPTIONS: See \link Residual_Map \endlink \n DEFAULT: RHO_RESIDUAL \ingroup Config*/
-  addEnumOption("RESIDUAL_FUNC_FLOW", Residual_Func_Flow, Residual_Map, RHO_RESIDUAL);
-  /*!\brief STARTCONV_ITER\n DESCRIPTION: Iteration number to begin convergence monitoring\n DEFAULT: 5 \ingroup Config*/
-  addUnsignedLongOption("STARTCONV_ITER", StartConv_Iter, 5);
-  /*!\brief CAUCHY_ELEMS\n DESCRIPTION: Number of elements to apply the criteria. \n DEFAULT 100 \ingroup Config*/
-  addUnsignedShortOption("CAUCHY_ELEMS", Cauchy_Elems, 100);
-  /*!\brief CAUCHY_EPS\n DESCRIPTION: Epsilon to control the series convergence \n DEFAULT: 1e-10 \ingroup Config*/
-  addDoubleOption("CAUCHY_EPS", Cauchy_Eps, 1E-10);
-  /*!\brief CAUCHY_FUNC_FLOW
-   *  \n DESCRIPTION: Flow functional for the Cauchy criteria \n OPTIONS: see \link Objective_Map \endlink \n DEFAULT: DRAG_COEFFICIENT \ingroup Config*/
-  addEnumOption("CAUCHY_FUNC_FLOW", Cauchy_Func_Flow, Objective_Map, DRAG_COEFFICIENT);
-  /*!\brief CAUCHY_FUNC_ADJFLOW\n DESCRIPTION: Adjoint functional for the Cauchy criteria.\n OPTIONS: See \link Sens_Map \endlink. \n DEFAULT: SENS_GEOMETRY \ingroup Config*/
-  addEnumOption("CAUCHY_FUNC_ADJFLOW", Cauchy_Func_AdjFlow, Sens_Map, SENS_GEOMETRY);
+  /*!\brief CONV_RESIDUAL_MINVAL\n DESCRIPTION: Min value of the residual (log10 of the residual)\n DEFAULT: -14.0 \ingroup Config*/
+  addDoubleOption("CONV_RESIDUAL_MINVAL", MinLogResidual, -14.0);
```

Comment:
This is wicked late in the PR process but the RESIDUAL_REDUCTION option is gone? It seems like the only residual convergence criteria is the CONV_RESIDUAL_MINVAL?

This might pose a significant problem since the actual value of the residual varies wildly between cases, solvers, non-dimensionalizations etc. The value of the residuals also vary between DIRECT and ADJOINT simulations. If running shape optimizations or anything that requires DIRECT and ADJOINT calculations in series, the same value of CONV_RESIDUAL_MINVAL would mean different levels of convergence.

Would it be terribly difficult to reinstate the RESIDUAL_REDUCTION option? I can help if needed.

Another option would be to normalize the residual value such that the initial residual for any case is 0 or 1.

Reply:
Using a relative residual is still possible, have a look here: https://su2code.github.io/docs/Guide-to-v7/

Unfortunately its not possible to choose different values for the adjoint and direct runs. We have to think about how we want to do that. There are also other options that may need different values for the adjoint (like the number of iterations). However, this PR does not remove any functionality regarding that.

Reply:
What I am gonna do for the moment is to make the `CONV_FIELD` option to accept a list of fields. If the field is not defined in the current solver it will simply be ignored (thats how I do it also when defining the fields for the history and volume output), The convergence criteria then applies to all available fields separately. This way you can at least define separate fields for adjoint and direct runs using the same config file.

Reply:
With the last commit I added the possibility of providing a list of fields for `CONV_FIELD`. They can be a mix of residuals and coefficients, e.g. `CONV_FIELD= RMS_DENSITY, DRAG`. If the field is a residual, `CONV_RESIDUAL_MINVAL` is applied. Otherwise, if it is a coefficient, the Cauchy options.

Right now it is *not* possible to choose different values for the convergence settings for each field. That will be too complicated in my opinion. However, to make the behavior of the Cauchy criterion a little bit more independent of the coefficient, I changed it to relative values. That means it is essentially now the relative difference between to consecutive iterations, averaged over the last N iterations. N can be set using the `CAUCHY_ELEMS` options. This should be much more intuitive. 

As mentioned in the comment above, if a field is not defined for the current solver, it will simply be ignored.

Reply:
Oh that's perfect. I especially like the ability to specify multiple CONV_FIELD options so that CAUCHY and RESIDUAL convergences can be applied at the same time. Thanks for that. 
I agree that having separate convergence criteria/iterations for DIRECT and ADJOINT calculations would be nice but complicated. A problem for another time
Sorry for not reading that guide fully before posting here. I didn't realize that the relative residual is a new field that could be used. On that note, is there an easy way to see the options available for CONV_FIELD (which I guess is the same as the list of output fields?) I understand that they change depending on the solver being used, but it'd be nice to see the options that would be needed to preserve behavior from the previous versions.

Reply:
> Just as a hint: To print a list of all possible output fields, use the `-d` flag, i.e. `SU2_CFD -d config.cfg`.

Every coefficient and residual listed there can be used for `CONV_FIELD`.

Reply:
Alright, I am just not a good reader then 🤦‍♂ 

Any chance that can be included in the custom output guide?